### DISCUSSION TITLE
How to prevent websocket from truncating messages on encountering 0-byte?

### DISCUSSION THREAD

**Discussion by DaveTJones** [uNetworking/uWebSockets#1493] [Q&A]
I'm trying to set up a server-client pair for a novel cryptographic key generation protocol I'm working with, using a uWebSockets server and WASM client. Everything works well, barring instances where the char I'm sending contains a 0. In these cases the 0 is interpreted as a terminating null-byte and the remainder of the message does not send. Is there any way to escape these 0's, or to set the length of the message ahead of sending such that the whole message is sent independant of its contents?

**Comment by hakoptak** [comment]
Hi Dave, you are sending data as text. If you send your data in the binary format then it should work fine. You define the data type when setting up your Websocket.

**Reply by uNetworkingAB** [reply]
A websocket can send (at any time, regardless of set up), messages as binary or text. Binary does not care about the content while text must be utf8 text. This is an argument to ws.send

**Reply by uNetworkingAB** [reply]
> You define the data type when setting up your Websocket.

This is not correct. There is no set up. You can send either, at any point regardless of set up.

**Reply by hakoptak** [reply]
Ah, my bad. Sorry for the confusion.
